Creating IPC HTTP classes

Archolic95 opened the pull request.

Archolic95 committed:
new GUID

Archolic95 committed:
http-get icon

Archolic95 committed:
HTTPs

garciadelcastillo commented:
Looking great Runjia! 

Comments:
- Let's put both HTTP components in the same level (primary?).
- For both HTTP components, can we set a default timeout value of 5000? 
- `POST` requests can have multiple mime types. Can we add a string `type` input, and have it by default on `application/json`?
- Also, can we set the default encoding to `UTF-8`?

Thanks!

garciadelcastillo commented:
Oh, also, let's not lose time with icons right now; just write some simple text on the icon, and we will find a volunteer to do the graphic design once the plugin is in better shape! :)

garciadelcastillo commented:
How are we doing on this front @runjiatian ?

Archolic95 committed:
Fix Level, Add Resource, Change Default Timeout

Archolic committed:
Encoding and Generic Types

Changed Default Encoding for HTTP-POST to UTF-8
Changed data input meme to generic

runjiatian commented:
All four comments resolved. 

- For meme types, currently I am using default string and **_Generic Type_**. Is there an explict set that you would prefer to restrict upon? Do we want to define a customized GH_Param to check whether input data belongs to this type set.

Archolic95 committed:
Adding WebSocket Component

Archolic95 committed:
minor fix
